File: 2018/07-jul/7-29-overloading/python/Animal.py
"""
Class representing an animal of a specific species
Author: Andrew Jarombek
Date: 7/28/2018
"""

from enum import Enum


class Animal(object):

    def __init__(self, name, species, description=""):
        """
        Construct an animal of a specific species with an optional description
        :param name: the name given to the animal
        :param species: specific species of the animal
        :param description: an optional description of the animal which defaults to
        an empty string
        """

        self.name = name
        self.species = species
        self.description = description


class AnimalSpecies(Enum):
    DEER = 1
    GROUNDHOG = 2
    RABBIT = 3
    RACCOON = 4
    SQUIRREL = 5
    CHIPMUNK = 6
    CROW = 7
    CARDINAL = 8

File: 2018/07-jul/7-29-overloading/javascript/Animal.js
/**
 * Class representing an animal of a specific species
 * @author Andrew Jarombek
 * @since 7/27/2018
 */
class Animal {

    /**
     * Constructor for an animal of a specific species with a name and optional description
     * of the animal
     * @param name - the name given to the animal
     * @param species - specific species of the animal
     * @param description - a description of the animal
     */
    constructor(name, species, description="") {
        this.name = name;
        this.species = species;
        this.description = description;
    }
}

Animal.Species = {
    DEER: 1,
    GROUNDHOG: 2,
    RABBIT: 3,
    RACCOON: 4,
    SQUIRREL: 5,
    CHIPMUNK: 6,
    CROW: 7,
    CARDINAL: 8
};

module.exports = Animal;